File: Java/Quicksort/Quicksort.java
import java.util.Arrays;

public class Quicksort {
    public static void main(String[] args) throws Exception {
        int[] numArray = {10, 7,8,9,1,5};
        quickSort(numArray, 0, numArray.length-1);
        System.out.println("Result: " + Arrays.toString(numArray));
        
    }
    
    public static int partition(int array[], int low, int high) {
        //pivot element to be placed at right position
        int pivot = array[high];

        //index of smaller element and indicates the right position of pivot
        //found so far
        int i = (low -1);
    
        for (int j = low; j <= high-1; j++) {
            //Checks if the current element is smaller than the pivot
            if(array[j] < pivot){
                //Increment index of smaller element
                i++;
                swap(array, i, j);
            }
        }
        swap(array, i+1, high);
        return (i+1);
    }

    //Swaps the two elements
    public static void swap(int[] array, int i, int j) {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }

    //Quick sort function
    public static void quickSort(int array[], int low, int high) {
        if(low < high){
            int partitionIndex = partition(array, low, high);
            //Sort the elements before partition and after partition
            quickSort(array, low, partitionIndex-1);
            quickSort(array, partitionIndex + 1, high);
        }
    }
}

File: Python/Quicksort/Quicksort.py
def partition(numArray, low, high):
	pivotIndex = low
	pivot = numArray[pivotIndex]
	while low < high:
		while low < len(numArray) and numArray[low] <= pivot:
			low += 1
		while numArray[high] > pivot:
			high -= 1
		if(low < high):
			numArray[low], numArray[high] = numArray[high], numArray[low]
	numArray[high], numArray[pivotIndex] = numArray[pivotIndex], numArray[high]
	return high
	
def quickSort(numArray, low, high):
	if (low < high):
		p = partition(numArray, low, high)
		quickSort(numArray, low, p - 1)
		quickSort(numArray, p + 1, high)
		
if __name__ == "__main__":
    numArray = [ 10, 7, 8, 9, 1, 5 ]
    quickSort(numArray, 0, len(numArray) - 1)
    for i in range(len(numArray)):
        print("%d" %numArray[i], end=" ")